app/javascript/mastodon/components/featured_carousel.tsx

```diff
@@ -0,0 +1,182 @@
+  const bind = useDrag(({ swipe: [swipeX] }) => {
+    handleSlideChange(swipeX * -1); // Invert swipe as swiping left loads the next slide.
+  });
+  const handlePrev = useCallback(() => {
+    handleSlideChange(-1);
+  }, [handleSlideChange]);
+  const handleNext = useCallback(() => {
+    handleSlideChange(1);
+  }, [handleSlideChange]);
+
+  // Handle wrapper height animation
+  const [wrapperStyles, wrapperSpringApi] = useSpring(() => ({
+    height: 100,
+  }));
+  const wrapperRef = useRef<HTMLDivElement>(null);
+  const wrapperEle = wrapperRef.current;
+  useEffect(() => {
+    if (!wrapperEle) {
+      return;
+    }
+    const currentSlideEle = wrapperEle.querySelector(
+      `[data-index="${slideIndex}"] > div`,
+    );
+
+    if (!currentSlideEle) {
+      return;
+    }
+    const height = currentSlideEle.scrollHeight;
+    if (!height) {
+      return;
+    }
+
+    void wrapperSpringApi.start({ height });
+  }, [slideIndex, wrapperSpringApi, wrapperEle]);
+
+  if (!accountId || pinnedStatuses.isEmpty()) {
+    return null;
+  }
+
+  return (
+    <div
+      className='featured-carousel'
+      {...bind()}
+      aria-roledescription='carousel'
+      aria-labelledby='featured-carousel-title'
+      role='region'
+    >
+      <div className='featured-carousel__header'>
+        <h4 className='featured-carousel__title' id='featured-carousel-title'>
+          {pinnedStatuses.size > 1 ? (
+            <FormattedMessage
+              id='featured_carousel.header.plural'
+              defaultMessage='Pinned Posts'
+            />
+          ) : (
+            <FormattedMessage
+              id='featured_carousel.header.single'
+              defaultMessage='Pinned Post'
+            />
+          )}
+        </h4>
+        {pinnedStatuses.size > 1 && (
+          <>
+            <IconButton
+              title={intl.formatMessage(messages.previous)}
+              icon='chevron-left'
+              iconComponent={ChevronLeftIcon}
+              onClick={handlePrev}
+            />
+            <FormattedMessage id='featured_carousel.post' defaultMessage='Post'>
+              {(text) => <span className='sr-only'>{text}</span>}
+            </FormattedMessage>
+            <span aria-live='polite'>
+              {slideIndex + 1} / {pinnedStatuses.size}
+            </span>
+            <IconButton
+              title={intl.formatMessage(messages.next)}
+              icon='chevron-right'
+              iconComponent={ChevronRightIcon}
+              onClick={handleNext}
+            />
+          </>
+        )}
+      </div>
+      <animated.div
+        className='featured-carousel__slides'
+        ref={wrapperRef}
+        style={wrapperStyles}
+        aria-atomic='false'
+        aria-live='polite'
+      >
+        {pinnedStatuses.map((statusId, index) => (
+          <animated.div
+            key={`f-${statusId}`}
+            style={{ x }}
+            className='featured-carousel__slide'
```

Comment:
You could probably handle the interactivity issue like this:
```suggestion
            inert={slideIndex !== index}
            className='featured-carousel__slide'
```

Reply:
i ended up doing this differently, as our version of React doesn't support `inert` so it errors with this code.